utils/httpstream.go

```diff
@@ -44,17 +45,38 @@ func StartHTTPServer(handler Handler, path, format string) {
 	}
 
 	// try to remove pre-existing UDS: ignore errors here
-	if tokens[0] == "unix" {
-		os.Remove(tokens[1])
+	phonySock := "/tmp/phonyfn.sock"
+	if uri.Scheme == "unix" {
+		os.Remove(phonySock)
 	}
 
-	listener, err := net.Listen(tokens[0], tokens[1])
+	listener, err := net.Listen(uri.Scheme, phonySock)
 	if err != nil {
-		panic("net.Listen error: " + err.Error())
+		log.Fatalln("net.Listen error: ", err)
+	}
+
+	if uri.Scheme == "unix" {
+		// somehow this is the best way to get a permissioned sock file, don't ask questions, life is sad and meaningless
+		f, err := os.Create(phonySock)
```

Comment:
hmm, os.Chmod() doesn't work?

Reply:
either works fine, just had a handle on the file since we need to make it so it was easier

Reply:
sounds good.

Reply:
oh, this is dumb actually. i don't know why this didn't click earlier (coffee and sustenance helps the brain, turns out)